aws-lc-rs/src/evp_pkey.rs

```diff
@@ -214,21 +243,196 @@ impl LcPtr<EVP_PKEY> {
         bytes: &[u8],
         evp_pkey_type: c_int,
     ) -> Result<Self, KeyRejected> {
+        #[cfg(not(feature = "fips"))]
+        if evp_pkey_type == EVP_PKEY_PQDSA {
+            return match bytes.len() {
+                2560 => Self::new(unsafe {
+                    EVP_PKEY_pqdsa_new_raw_private_key(NID_MLDSA44, bytes.as_ptr(), bytes.len())
+                }),
+                4032 => Self::new(unsafe {
+                    EVP_PKEY_pqdsa_new_raw_private_key(NID_MLDSA65, bytes.as_ptr(), bytes.len())
+                }),
+                4896 => Self::new(unsafe {
+                    EVP_PKEY_pqdsa_new_raw_private_key(NID_MLDSA87, bytes.as_ptr(), bytes.len())
+                }),
+                _ => Err(()),
+            }
+            .map_err(|()| KeyRejected::unspecified());
+        }
+
         Self::new(unsafe {
             EVP_PKEY_new_raw_private_key(evp_pkey_type, null_mut(), bytes.as_ptr(), bytes.len())
         })
-        .map_err(|()| KeyRejected::invalid_encoding())
+        .map_err(|()| KeyRejected::unspecified())
     }
 
     pub(crate) fn parse_raw_public_key(
         bytes: &[u8],
         evp_pkey_type: c_int,
     ) -> Result<Self, KeyRejected> {
+        #[cfg(not(feature = "fips"))]
+        if evp_pkey_type == EVP_PKEY_PQDSA {
+            return match bytes.len() {
+                1312 => Self::new(unsafe {
+                    EVP_PKEY_pqdsa_new_raw_public_key(NID_MLDSA44, bytes.as_ptr(), bytes.len())
+                }),
+                1952 => Self::new(unsafe {
+                    EVP_PKEY_pqdsa_new_raw_public_key(NID_MLDSA65, bytes.as_ptr(), bytes.len())
+                }),
+                2592 => Self::new(unsafe {
+                    EVP_PKEY_pqdsa_new_raw_public_key(NID_MLDSA87, bytes.as_ptr(), bytes.len())
+                }),
+                _ => Err(()),
+            }
+            .map_err(|()| KeyRejected::unspecified());
+        }
+
         Self::new(unsafe {
             EVP_PKEY_new_raw_public_key(evp_pkey_type, null_mut(), bytes.as_ptr(), bytes.len())
         })
         .map_err(|()| KeyRejected::invalid_encoding())
     }
+
+    pub(crate) fn sign<F>(
+        &self,
+        message: &[u8],
+        digest: Option<&'static digest::Algorithm>,
+        padding_fn: Option<F>,
```

Comment:
Genuinely curious, is there a particular reason for the workaround and why we can't do `Option<*mut EVP_PKEY_CTX>`? Is it due to Option not being able to take mutable pointers?